base/file.jl

```diff
@@ -671,41 +671,87 @@ struct uv_dirent_t
 end
 
 """
-    readdir(dir::AbstractString=".") -> Vector{String}
+    readdir(dir::AbstractString=pwd(); join::Bool=true) -> Vector{String}
```

Comment:
I fixed `join=true` to `join=false` (skipping making a PR hoping it would save CI some work, but CI got triggered anyway...)
Also, why did you choose `pwd()` for `dir`'s default? I guess it makes `readdir(join=true)` more useful, as otherwise it would be quite equivalent to `readdir()` (modulo the leading dot), but I think I would still expect `.` rather than `pwd()` as the default, as is the case for quite many command line tools.

Reply:
> Also, why did you choose `pwd()` for `dir`'s default? I guess it makes `readdir(join=true)` more useful, as otherwise it would be quite equivalent to `readdir()` (modulo the leading dot), but I think I would still expect `.` rather than `pwd()` as the default, as is the case for quite many command line tools.

Yes, I did that so that `readdir(join=true)` would return the absolute paths of everything in the current directory. What difference does using `"."` versus `pwd()` for the default make? The only case I can think of where it would matter is if the current working directory has been deleted. Then `readdir(".")` might work but `pwd()` would fail. Yeah, now that I write that (and tested the difference), I guess that's what we should do: use `"."` as the default when `join` is false and `pwd()` as the default when `join` is true.

Reply:
>  What difference does using "." versus pwd() for the default make?

I hadn't even thought about deleted directories, but I meant that I would expect `readdir(join=true)` to return relative entries of the form `./this.txt`. But I'm not convinced either way actually, because it's probably way more common to want the absolute path when you ask `join=true`.

Reply:
Returning a bunch of names with `./` prefixed is pretty useless, whereas returning the absolute paths of all the files in the current directory is useful. If you really want the former, you can always call it as `readdir(".", join=true)`. If you haven't asked for `join=true` then the only difference between `pwd()` and `"."` as a default is the deleted directory thing, which I'm working on a PR to fix.

Reply:
I must admit to not really understand why I argued this point, but I'm glad it allowed you to find the "deleted directory" corner case :)

Reply:
I had already thought of it but found it too obscure to bother with, but why not be as thorough as possible: https://github.com/JuliaLang/julia/pull/33175.